DuckDuckGo/NewTabPageViewController.swift

```diff
@@ -174,3 +196,45 @@ extension NewTabPageViewController: HomeScreenTransitionSource {
         view
     }
 }
+
+extension NewTabPageViewController {
+
+    func showNextDaxDialogNew(dialogProvider: NewTabDialogSpecProvider, factory: any NewTabDaxDialogProvider) {
+        dismissHostingController(didFinishNTPOnboarding: false)
+
+        guard let spec = dialogProvider.nextHomeScreenMessageNew() else { return }
+
+        let onDismiss = {
+            dialogProvider.dismiss()
+            self.dismissHostingController(didFinishNTPOnboarding: true)
+        }
+        let daxDialogView = AnyView(factory.createDaxDialog(for: spec, onDismiss: onDismiss))
+        let hostingController = UIHostingController(rootView: daxDialogView)
+        self.hostingController = hostingController
+
+        hostingController.view.backgroundColor = .clear
+        addChild(hostingController)
+        view.addSubview(hostingController.view)
+        hostingController.view.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            hostingController.view.topAnchor.constraint(equalTo: view.topAnchor),
+            hostingController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            hostingController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            hostingController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+
+        hostingController.didMove(toParent: self)
+
+        newTabPageModel.startOnboarding()
+    }
```

Comment:
Copied the code from `HomeViewController+DaxDialogs`, but included `addChild(hostingController)` which was missing in the original.

Reply:
Included this call in the HomeViewController code (https://github.com/duckduckgo/iOS/pull/3203/commits/cd424a9321f6dd2dd6eec1485b03cf7fda7084d3).